pkg/pipeline/PipelineStageService.go

```diff
@@ -1417,29 +1475,50 @@ func (impl *PipelineStageServiceImpl) DeleteCiStage(stageReq *bean.PipelineStage
 //DeleteCiStage and related methods starts
 
 // BuildPrePostAndRefPluginStepsDataForWfRequest and related methods starts
-func (impl *PipelineStageServiceImpl) BuildPrePostAndRefPluginStepsDataForWfRequest(ciPipelineId int) ([]*bean.StepObject, []*bean.StepObject, []*bean.RefPluginObject, error) {
-	//get all stages By ciPipelineId
-	ciStages, err := impl.pipelineStageRepository.GetAllCiStagesByCiPipelineId(ciPipelineId)
+func (impl *PipelineStageServiceImpl) BuildPrePostAndRefPluginStepsDataForWfRequest(pipelineId int, stageType string) ([]*bean.StepObject, []*bean.StepObject, []*bean.RefPluginObject, error) {
+	//get all stages By pipelineId (it can be ciPipelineId or cdPipelineId)
+	var pipelineStages []*repository.PipelineStage
+	var err error
+	if stageType == ciEvent {
+		pipelineStages, err = impl.pipelineStageRepository.GetAllCiStagesByCiPipelineId(pipelineId)
+	} else {
+		//cdEvent
+		pipelineStages, err = impl.pipelineStageRepository.GetAllCdStagesByCdPipelineId(pipelineId)
+	}
 	if err != nil && err != pg.ErrNoRows {
-		impl.logger.Errorw("error in getting all ci stages by ciPipelineId", "err", err, "ciPipelineId", ciStages)
+		impl.logger.Errorw("error in getting all ci stages by pipelineId", "err", err, "pipelineId", pipelineId, "stageType", stageType)
 		return nil, nil, nil, err
 	}
 	var preCiSteps []*bean.StepObject
 	var postCiSteps []*bean.StepObject
+	var preCdSteps []*bean.StepObject
+	var postCdSteps []*bean.StepObject
 	var refPluginsData []*bean.RefPluginObject
 	var refPluginIds []int
-	for _, ciStage := range ciStages {
+	for _, pipelineStage := range pipelineStages {
 		var refIds []int
-		if ciStage.Type == repository.PIPELINE_STAGE_TYPE_PRE_CI {
-			preCiSteps, refIds, err = impl.BuildCiStageDataForWfRequest(ciStage)
+		if pipelineStage.Type == repository.PIPELINE_STAGE_TYPE_PRE_CI {
+			preCiSteps, refIds, err = impl.BuildPipelineStageDataForWfRequest(pipelineStage)
 			if err != nil {
-				impl.logger.Errorw("error in getting pre ci steps data for wf request", "err", err, "ciStage", ciStage)
+				impl.logger.Errorw("error in getting pre ci steps data for wf request", "err", err, "ciStage", pipelineStage)
 				return nil, nil, nil, err
 			}
-		} else if ciStage.Type == repository.PIPELINE_STAGE_TYPE_POST_CI {
-			postCiSteps, refIds, err = impl.BuildCiStageDataForWfRequest(ciStage)
+		} else if pipelineStage.Type == repository.PIPELINE_STAGE_TYPE_POST_CI {
```

Comment:
these ``if else`` blocks need refactoring

Reply:
refactored it, now using switch in place of if..else